### ISSUE TITLE
We have DiscussionForumPosting but no DiscussionForum...

### ISSUE THREAD

**Issue by danbri** [schemaorg/schemaorg#2016]
Early on, we had some discussion around modeling Forums better, but it seems we don't even have DiscussionForum, despite having DiscussionForumPost. Should we add that at least?

**Comment by vickitardif** [comment]
We also have SocialMediaPosting, but not SocialMediaPlatform.

+1 to adding both.

**Comment by danbri** [comment]
We also have Question Answer, QAPage etc... ... are Q/A sites socialmedia or discussion forums? What about fact checking sites with a community element like Snopes? ...

**Comment by thadguidry** [comment]
I would classify these sites as discussion forums

Quora.com Stackoverflow.com 

The old mythbusting, urban legend debunking sites...have now all converted to typically being classified as Fact Checking sites.  (for the lay person, true or false, fact or fiction, etc)

The Snopes.com web site was founded by David Mikkelson, a project begun in 1994 as an expression of his interest in researching urban legends that has since grown into the oldest and largest fact-checking site on the Internet 

Here's an older list of some of the most popular still.  I classify them as Fact Checking, but if you limited Fact Checking to only News worthy....I can only tell you that the latest Hoax sometimes quickly becomes national news worthy and then someone "fact checks" an Internet Hoax.  Again, its all about "true or not" which has surfaced into our new lingua franca of "Fact Check sites" no matter their subdomain or speciality areas.
https://www.techrepublic.com/blog/10-things/top-10-sites-to-debunk-internet-hoaxes/

**Comment by Meteor0id** [comment]
@vholland

Defining a SocialMediaPlatform is a weird way to seperate social media from the rest of the world wide web. that would aknoledge that social media platforms are a different thing than websites, which they are not. Social Media websites have no elevated status, they are no sperate thing from a website with a blog on it or a website with a forum on it.

You could of course make a SocialMediaPage as a spesific type of WebPage. Just like a Blog is a spesific type of WebPage.


The same for DicsussionForum. It makes little sense to adress the forum as a whole. There is a Website as a whole, and there are WebPages as part of it. those WebPages may have a spesific role, such as being a forum page, a blog page, a news page.. but lets not act like a forum or collection of social media posts is a thing by itself.

**Comment by vickitardif** [comment]
I respectfully disagree. I don't think most users view Twitter or Facebook as just a website.

**Comment by Meteor0id** [comment]
@vholland I agree most users do not view Twitter or Facebook as just a website. But that is also why I am objecting against creating a whole seperate type for them. At this age, Google Facebook and Twitter have become a sort of autority on how interaction over the world wide web should take place, that makes sense today, but some serious compeditors might come around in the upcoming years. Lets not forget that they are still just websites, be it very succesful ones. So lets not come up with vocab now which aknoledges them as something seperate from other websites. Their name already identiefies them as the major authority they are, so they do not require a seperate Type for that. Discussionforums and SocialMediaPLatforms are a type of webpage.

**Comment by vickitardif** [comment]
@Meteor0id I think we are closer than may appear. I don't think any site owns the type SocialMediaPlatform and it should apply to more than the big players. If an upstart comes along with 3 users chatting, they should feel just as comfortable calling themselves a SocialMediaPlatform.

**Comment by danbri** [comment]
@vholland @Meteor0id I would prefer to avoid terminology like "platform" which carries heavy and potentially conversial baggage ([e.g. article](https://medium.com/@subsign/is-facebook-a-platform-or-a-publisher-f2e2fd04d4eb)). While "discussion forum" (or the endearingly archaic "website") is less catchy, it's also less likely to be a distinction appealed to in legalistic settings. Which I'd count as a plus.

**Comment by Demeter** [comment]
@danbri I was just reading about Google Dataset Search, and ended up here. I read and [responded to](https://medium.com/@ellieaskswhy/a-publisher-is-more-than-just-a-company-or-person-that-curates-and-distributes-content-d29e226189fe) the article you linked to. We are mutuals on Twitter, so I hope you will indulge my barging in to your taxonomy discussion despite the fact that I am not a schema dot org contributor.

I strongly agree with @vholland about using type SocialMediaPlatform. Yes, Facebook is a mess, as the article you cited mentioned, because Z is being both publisher and platform. Regardless, it is important to remember that what might seem the dominant players now, as @Meteor0id said, were not always so, and will not necessarily remain so. 

Type = SocialMediaPlatform can be applied to Twitter, Facebook, Youtube, and Google+ as well as Gab.ai, diaspora.org, StockTwits, Microsoft TechNet, gnu.io/social/ aka www.gnu.org/software/social/ and (Evan Prodromou's?) departed but not forgotten identi.ca that became StatusNet. As @vholland said, 

> If an upstart comes along with 3 users chatting, they should feel just as comfortable calling themselves a SocialMediaPlatform.

 The modest Gnu social website, sealion.club, that I frequent is an example of such.

Final thought: Any social media website that has a developer ecosystem, (criteria could be an active developer forum?) could be considered type=SocialMediaPlatform. That includes Facebook, Twitter, Google+, Youtube, Gnu social/statusnet, diaspora, reddit, the former Digg, Social.Technet.Microsoft.com and possibly others.

**Comment by eSilverStrike** [comment]
The description of DisccussionForumPosting is "A posting to a discussion forum.". 

A typical forum is made up of categories. Categories contain forums, and a forum contains topics. A topic contains 1 or more forum posts.

The description seems to indicate that this schema is for an individual forum post but the example seems to show it is for a topic which can have multiple forum posts from multiple authors.
